src/windows.jl

```diff
@@ -123,16 +123,16 @@ function bartlett(n::Integer)
 end
 
 """
-    gaussian(n, sigma)
+    gaussian(n, σ)
 
-Gaussian window of length `n` parameterized by the standard
-deviation `sigma`.
+Gives an n-sample gaussian window defined by sampling the function
+\$w(x) = e^{-\\frac 1 2 \\left(\\frac x σ \\right)^2}\$ in the range
+\$[-0.5,0.5]\$. This means that for \$σ=0.5\$ the endpoints of the window will
+correspond to 1 standard deviation away from the center.
 """
-function gaussian(n::Integer, sigma::Real)
-    if !(0 < sigma <= 0.5)
-        error("sigma must be greater than 0 and less than or equal to 0.5.")
-    end
-    [exp(-0.5*((k-(n-1)/2)/(sigma*(n-1/2)))^2) for k=0:(n-1)]
+function gaussian(n::Integer, σ::Real)
+    σ > 0.0 || error("σ must be positive")
+    [exp(-0.5*((k-(n-1)/2)/(σ*(n-1)))^2) for k=0:(n-1)]
```

Comment:
I realize this is preexisting, but note that 0.5 is a `Float64` literal, so this will unconditionally promote everything to `Float64`. It would be safer (in some sense) to simply divide by 2.

Reply:
we'll already be in `Float64` land because of the `(k-(n-1)/2)`

Reply:
If we use `[exp(-((2*k-(n-1))/(2*σ*(n-1)))^2 / 2) for k=0:(n-1)]`, one could get a `Float32`-valued window by specifying σ as a `Float32`. But the other window functions also just return `Float64`s, so I doubt it's worth it.